zrml/market-commons/src/lib.rs

```diff
@@ -123,6 +124,14 @@ mod pallet {
             }
         }
 
+        fn market_iter() -> Box<
+            dyn Iterator<
+                Item = (Self::MarketId, Market<Self::AccountId, Self::BlockNumber, Self::Moment>),
+            >,
+        > {
```

Comment:
Hmm, yeah, my solution is dumb. Thanks for the suggestion. I wanted to specify the return type in terms of the trait that it implements. `impl Iterator<...>` doesn't work because we're implementing a method here. So I used a `Box`. For some reason, I forgot that I could just specify the exact return type.